attestation-agent/attestation-agent/src/eventlog.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{fs::File, io::Write, path::PathBuf};
+
+use anyhow::{anyhow, Context, Result};
+
+pub struct EventLog {
+    file: File,
+}
+
+impl EventLog {
+    pub fn new(path: &str) -> Result<Self> {
+        let eventlog_path = PathBuf::from(path);
+        let parent_dir = eventlog_path
+            .parent()
+            .ok_or_else(|| anyhow!("illegal eventlog path: cannot locate the parent dir"))?;
+
+        std::fs::create_dir_all(parent_dir).context("create eventlog parent dir")?;
+        let file = File::create(path).context("create eventlog")?;
+        Ok(Self { file })
+    }
+
+    pub fn write_log(&mut self, log: &[u8]) -> Result<()> {
```

Comment:
Just wonder, why using `u8` instead of `String` (UTF-8 encoding) as the log entry type?

Reply:
There is no any special reasons. If you prefer, I can turn it into `String`